### ISSUE TITLE
Error: Cannot find module for page: /_document

### ISSUE THREAD

**Issue by kmalakoff** [vercel/next.js#31636]
### What version of Next.js are you using?

12.0.4

### What version of Node.js are you using?

16.13.0

### What browser are you using?

Chrome

### What operating system are you using?

macOS

### How are you deploying your application?

none

### Describe the Bug

When running "next build" in an application with a _document.js page, the build fails.

The default doesn't work from this example: https://nextjs.org/docs/messages/missing-document-component
```
import Document, { Html, Head, Main, NextScript } from 'next/document'

export default class MyDocument extends Document {
  render() {
    return (
      <Html>
        <Head />
        <body>
          <Main />
          <NextScript />
        </body>
      </Html>
    )
  }
} 
```

I get errors like:

```
info  - Collecting page data ...
Error: Cannot find module for page: /_document
    at pageNotFoundError (/path-to-next-app/node_modules/next/dist/server/require.js:16:17)
    at getPagePath (/path-to-next-app/node_modules/next/dist/server/require.js:40:15)
    at Object.requirePage (/path-to-next-app/node_modules/next/dist/server/require.js:45:22)
    at Object.loadComponents (/path-to-next-app/node_modules/next/dist/server/load-components.js:56:23)
    at Object.getNamedExports (/path-to-next-app/node_modules/next/dist/build/utils.js:680:51)
    at execFunction (/path-to-next-app/app-web/node_modules/jest-worker/build/workers/processChild.js:145:17)
    at execHelper (/path-to-next-app/app-web/node_modules/jest-worker/build/workers/processChild.js:124:5)
    at execMethod (/path-to-next-app/app-web/node_modules/jest-worker/build/workers/processChild.js:128:5)
    at process.messageListener (/path-to-next-app/app-web/node_modules/jest-worker/build/workers/processChild.js:46:7)
    at process.emit (node:events:390:28) {
  type: 'Error',
  code: 'ENOENT'
}

```

I cannot seem to find an existing example or migration guide for Next 12 with a custom _document and have reviewed issues since Next 12 was related to _document, but am unsure if I'm supposed to make changes or if the _document approach is in flux so the build is no longer being tested / maintained.

### Expected Behavior

Successful build

### To Reproduce

1. Add a _document.js file to the pages folder like:

```
import Document, { Html, Head, Main, NextScript } from 'next/document'

export default class MyDocument extends Document {
  render() {
    return (
      <Html>
        <Head />
        <body>
          <Main />
          <NextScript />
        </body>
      </Html>
    )
  }
}
```

2. run "next build"

**Comment by timneutkens** [comment]
I'm unable to reproduce this based on the provided reproduction, could you provide a repository?

**Comment by kmalakoff** [comment]
I wasn't able to strip down my application to a minimal reproducible version nor create a failing app from scratch. 

After studying similar next issues, it seems like "Cannot find module for page: /_document" does not actually describe a root cause, but is a symptom of a general failure so I started a new project and migrated my app to it in steps to try to figure out where an error was introduced. After migrating my full app, no errors were introduced and "next build" worked successfully so I'm unblocked.

**Comment by hlolli** [comment]
I'm getting this as well after an update to 12.0.4, mysterious to me, definitely unrelated to the error message and stack trace

```console
Error: Cannot find module '/project/.next/server/pages/_document.js'
Require stack:
- /project/node_modules/next/dist/server/require.js
- /project/node_modules/next/dist/server/load-components.js
- /project/node_modules/next/dist/build/utils.js
- /project/node_modules/next/dist/build/output/store.js
- /project/node_modules/next/dist/build/output/index.js
- /project/node_modules/next/dist/cli/next-dev.js
- /project/node_modules/next/dist/bin/next
```

**Comment by hlolli** [comment]
I found a quick fix by adding NODE_PATH:

```
NODE_PATH=$(pwd)/node_modules next dev
```

I wonder if nextjs is overriding node_path internally when it shouldn't, the esm/cjs story on top of the old pirates's require-steal is the source of many evils.

**Comment by balazsorban44** [comment]
This issue has been automatically locked due to no recent activity. If you are running into a similar issue, please create a new issue with the steps to reproduce. Thank you.
